### PR TITLE
Retry internal id lookup when not found #2533

### PR DESCRIPTION
**Pull request by albinsuresh**
## Proposed changes

Even if the internal ID lookup fails on the initial attempt (presumably because of delays in the creation of the device), retry a few more times before failing.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/extensions/c8y_http_proxy/src/actor.rs [hunk 1]
```diff
@@ -192,37 +193,55 @@ impl C8YHttpProxyActor {
...
-        let res = match self.peers.http.await_response(request).await? {
-            Ok(response) => match response.status() {
-                StatusCode::OK => Ok(Ok(response)),
-                StatusCode::UNAUTHORIZED | StatusCode::FORBIDDEN => {
-                    self.get_fresh_token().await?;
-                    let request = HttpRequestBuilder::get(&url_get_id)
-                        .bearer_auth(self.end_point.token.clone().unwrap_or_default())
-                        .build()?;
-                    Ok(self.peers.http.await_response(request).await?)
+        let attempt = 1;
```
Comment (rina23q) [inline crates/extensions/c8y_http_proxy/src/actor.rs:201]: Where do you increase this `attempt`...?

Comment (albinsuresh) [reply crates/extensions/c8y_http_proxy/src/actor.rs:201]: Good catch ;)

**Review by rina23q — APPROVED** [review]
LGTM

**Review by didier-wenzek — APPROVED** [review]
Approved

**Pull request merged**